File: java-version/src/week3/leetcode179.java
package week3;
//                https://leetcode.com/problems/largest-number/
//                        179. Largest Number


import week1.day4.leetcode1337;

import java.util.Arrays;
import java.util.Comparator;

public class leetcode179 {
     static String largestNumber(int[] nums) {
        String[] asStrs = new String[nums.length];
        for (int i = 0; i < nums.length; i++) {
            asStrs[i] = String.valueOf(nums[i]);
        }

         Arrays.sort(asStrs, new Comparator<String>() {
             @Override
             public int compare(String a, String b) {
                 String order1 = a + b;
                 String order2 = b + a;
                 return order2.compareTo(order1);
             }
         });
         if (asStrs[0].equals("0")) {
             return "0";
         }
         StringBuilder largestNumberStr = new StringBuilder();
        for (String str : asStrs) {
        largestNumberStr.append(str);
    }

        return largestNumberStr.toString();


}


    public static void main(String[] args) {
        int[] nums = {3, 30, 34, 5, 9};
        System.out.println(largestNumber(nums));
    }
    }

File: codingTest/3week-assignment/day1/leetcode179.py
# https://leetcode.com/problems/largest-number/description/
# 179. Largest Number
from typing import List


class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        lst = []

        for ele in nums:
            lst += [str(ele)]

        n = len(lst)

        for i in range(n):
            for j in range(i + 1, n):

                if str(lst[i]) + str(lst[j]) > str(lst[j]) + str(lst[i]):
                    continue
                else:
                    lst[i], lst[j] = lst[j], lst[i]

        ans = ''.join(lst)

        if int(ans) == 0:
            return "0"

        return ans

